### DISCUSSION TITLE
Timeseries: Supply predict_models with own X (exogenous ) Variables instead of using holdout

### DISCUSSION THREAD

**Discussion by ConnorReed246** [pycaret/pycaret#2833] [Q&A]
Hi,
I am trying to use predicted exogenous to improve forecast quality, to get a measure of how much this helps in a certain case I need to simulate predicted exogenous values for the holdout set aswell before finalizing on the whole dataset. 

But before finalizing predict_model uses it's own holdout set, which can't be modified with set_config, even if using the X parameter. Which seems to make it impossible to get the best of both worlds where:

a) Before finalizing we use the predicted Values to get an accurate score
b) After finalizing we accutally use the real values up to the point we know them

Any Ideas?

Thanks,
Connor

**Comment by ngupta23** [comment]
> Before finalizing we use the predicted Values to get an accurate score

This is not correct. Before finalizing, you use the actual values that were provided in the dataset. There is no prediction done for exogenous variables inside the library.

Are you asking for the ability to use predicted holdout values before finalizing? If so, how would you provide the same for cross validation?
